src/main/scala/org/deepdive/inference/datastore/SQLInferenceDataStore.scala

```diff
@@ -656,41 +432,102 @@ trait SQLInferenceDataStore extends InferenceDataStore with Logging {
     }
 
     // dump factor meta data
-    du.unload(s"dd_factormeta", s"${groundingPath}/dd_factormeta", dbSettings, parallelGrounding,
-      s"SELECT * FROM ${FactorMetaTable}")
+    val groundingDir = getFileNameFromPath(groundingPath)
+    du.unload(InferenceNamespace.getFactorMetaFileName, 
+      s"${groundingPath}/${InferenceNamespace.getFactorMetaFileName}", 
+      dbSettings, parallelGrounding, s"SELECT * FROM ${FactorMetaTable}",
+      groundingDir)
+  }
+
+  // create feature stats for boolean LR function
+  def createFeatureStats(factorDesc: FactorDesc, querytable: String, weightlist: String,
+    weightDesc: String) {
+    // Create feature statistics support tables for error analysis, 
+    // only if it's boolean LR feature (the most common one)
+    if (factorDesc.func.variables.length == 1 && factorDesc.func.variableDataType == "Boolean") {
+      // This should be a single variable, e.g. "is_true"
+      val variableName = factorDesc.func.variables.map(v => 
+          s""" ${quoteColumn(v.toString)} """).mkString(",")
+      val groupByClause = weightlist match {
+        case "" => ""
+        case _ => s"GROUP BY ${weightlist}"
+      }
+      execute(s"""
+      INSERT INTO ${FeatureStatsSupportTable}
+      SELECT ${weightDesc} as description,
+             count(CASE WHEN ${variableName}=TRUE THEN 1 ELSE NULL END) AS pos_examples,
+             count(CASE WHEN ${variableName}=FALSE THEN 1 ELSE NULL END) AS neg_examples,
+             count(CASE WHEN ${variableName} IS NULL THEN 1 ELSE NULL END) AS queries
+      FROM ${querytable}
+      ${groupByClause};
+      """)
+      execute(analyzeTable(FeatureStatsSupportTable))
+    }
+  }
+
+  // convert grounding file format to be compatible with sampler
+  // for more information about format, please refer to deepdive's website
+  def convertGroundingFormat(groundingPath: String) {
+    log.info("Converting grounding file format...")
+    val ossuffix = if (System.getProperty("os.name").startsWith("Linux")) "linux" else "mac"
+    // TODO: this python script is dangerous and ugly. It changes too many states!
+    val cmd = s"python ${Context.deepdiveHome}/util/tobinary.py ${groundingPath} " + 
+        s"${Context.deepdiveHome}/util/format_converter_${ossuffix} ${Context.outputDir}"
+    log.debug("Executing: " + cmd)
+    val exitValue = cmd!(ProcessLogger(
+      out => log.info(out),
+      err => log.info(err)
+    ))
+
+    exitValue match {
+      case 0 => 
+      case _ => throw new RuntimeException("Converting format failed.")
+    }
+  }
+
+  def groundFactorsAndWeights(factorDescs: Seq[FactorDesc],
+    calibrationSettings: CalibrationSettings, du: DataLoader,
+    dbSettings: DbSettings, groundingPath: String, parallelGrounding: Boolean,
+    skipLearning: Boolean, weightTable: String) {
+    val groundingDir = getFileNameFromPath(groundingPath)
+
+    // save last weights
+    if (skipLearning && weightTable.isEmpty()) {
+      execute(copyLastWeightsSQL)
```

Comment:
If this clause is executed, does that mean that you will copy the last weights, and then immediately drop the destination table?  Not clear what the desired behavior is here.